Fix GIF parser

ojii opened the pull request:
Previously, the GIF parser would assume a GIF is animated if it had the Netscape Application Extension. This can lead to false positives where a GIF can have all the indicators of an animated GIF but only one Image Descriptor.

Now, the code checks for the absence of the Graphic Control Extension before an Image Descriptor to mean non-animated and otherwise checks if there are multiple Image Descriptors. This unfortunately requires parsing much further into the data stream.

The example cli now also supports specifiying the buffer size.

Fixes #7

ojii committed:
Fix GIF parser

Previously, the GIF parser would assume a GIF is animated if it had the
Netscape Application Extension. This can lead to false positives where a
GIF can have all the indicators of an animated GIF but only one Image
Descriptor.

Now, the code checks for the absence of the Graphic Control Extension
before an Image Descriptor to mean non-animated and otherwise checks if
there are multiple Image Descriptors. This unfortunately requires
parsing much further into the data stream.

The example cli now also supports specifiying the buffer size.

FurqanHabibi commented on README.md:
````diff
@@ -30,7 +30,8 @@ else:
 ```
 
-You should not pass the entire image data, the first kilobyte or so should suffice.
+You should not pass the entire image data, the first kilobyte or so should suffice for most formats, other than GIF
+where a larger amount of data may be required to determine whether the image is animated or not.
````
Should we add example of supplying the buf_size to the snippet above?

FurqanHabibi commented on src/gif.rs:
```diff
@@ -1,43 +1,107 @@
+                    found_image = true;
+                    cursor.seek(SeekFrom::Current(8))?;
+                    let flags = cursor.read_u8()?;
+                    if let Some(size) = color_table_size(flags) {
+                        cursor.seek(SeekFrom::Current(size))?;
```
:nit:
```suggestion
                    if let Some(size) = color_table_size(flags) {
                        // skip Local Color Table
                        cursor.seek(SeekFrom::Current(size))?;
```

FurqanHabibi reviewed (approved):
The parsing part LGTM.
Ref : https://www.w3.org/Graphics/GIF/spec-gif89a.txt

ojii committed:
test for single-image gif with AE/GCE

Usually, the "NETSCAPE2.0" Application Extension is used for looping
GIFs and the Graphic Control Extension is used for animated GIFs.
However, such files can still have only a single image, resulting in a
non-animated GIF. While rather nonsensical, these files should still be
handled correctly, so one such file was crafted for testing.
